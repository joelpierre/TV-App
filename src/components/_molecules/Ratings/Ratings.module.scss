.Ratings {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  text-decoration: none;

  &__star {
    margin-right: (spacing(0.5) / 2);
  }

  &__figures {
    margin-left: spacing();
    font-family: $font-bold;
    color: inherit;
  }
}
