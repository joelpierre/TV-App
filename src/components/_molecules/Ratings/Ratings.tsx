import React from 'react';
import classNames from 'classnames';

import styles from './Ratings.module.scss';
import Star from '@atoms/Star';

interface IRatingsProps {
  className?: string;
  averageRating: number | null;
  withFigures?: boolean;
}

const Ratings: React.FunctionComponent<IRatingsProps> = ({
  className,
  averageRating,
  withFigures = false
}) => {
  const starRange = Array.from({ length: 5 }, (_, starItem) => starItem);
  const ratingOutOfFive = averageRating ? averageRating / 2 : 0;
  const roundedRating = Math.floor(ratingOutOfFive);

  return (
    <div className={classNames(styles.Ratings, className)}>
      {starRange.map((index) => {
        const starPosition = index + 1;
        return <Star key={index} className={styles.Ratings__star} isActive={starPosition <= roundedRating} />;
      })}

      {withFigures && (
        <span className={styles.Ratings__figures}>
          {ratingOutOfFive === 0 ? '(no rating)' : `${ratingOutOfFive}/5`}
        </span>
      )}
    </div>
  );
};

export default Ratings;
