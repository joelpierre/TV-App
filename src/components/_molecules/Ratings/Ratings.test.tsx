import React from 'react';
import { render, RenderResult } from 'test-utils';

import Ratings, { IRatingsProps } from './Ratings';

const defaultProps: IRatingsProps = {
  averageRating: 5
};

const setup = (props: Partial<IRatingsProps> = {}): RenderResult => {
  const setupProps = { ...defaultProps, ...props };
  return render(<Ratings {...setupProps} />);
};

describe('<Ratings/>', () => {
  let wrapper: RenderResult;

  beforeEach(() => {
    wrapper = setup();
  });

  it('Should render without crashing', () => {
    expect(wrapper.container).not.toBeEmptyDOMElement();
  });

  it('should not have the figures visible', () => {
    expect(wrapper.queryByTestId('figures')).toBeFalsy();
  });

  it('should match snapshot', () => {
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  describe('Given "withFigures" is true', () => {
    it('Should return ratings with the figures', () => {
      wrapper = setup({ withFigures: true });
      expect(wrapper.getByTestId('figures')).toBeTruthy();
    });
  });
});
