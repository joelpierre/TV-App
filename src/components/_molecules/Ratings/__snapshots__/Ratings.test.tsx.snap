// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ratings/> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Ratings"
  >
    <span
      class="Star Ratings__star Star--is-active"
    />
    <span
      class="Star Ratings__star Star--is-active"
    />
    <span
      class="Star Ratings__star"
    />
    <span
      class="Star Ratings__star"
    />
    <span
      class="Star Ratings__star"
    />
  </div>
</DocumentFragment>
`;
